addons/test_mail/tests/test_mail_composer.py

```diff
@@ -453,6 +453,37 @@ def test_mail_composer_rights_attachments(self):
             sorted(self.test_record.message_ids[0].attachment_ids.mapped('name')),
             sorted(template_1_attachment_name))
 
+    @users('employee')
+    def test_mail_composer_template_switching(self):
+        """ Ensure that the current user's identity serves as the sender,
+        particularly when transitioning from a template with a designated sender to one lacking such specifications.
+        Moreover, we verify that switching to a template lacking any subject maintains the existing subject intact.
+        """
+        # Setup: Prepare Templates
+        template_with_sender_subject = self.template.copy({
+            'email_from': 'not_current_user@template_with_sender_subject.com',
+            'subject': "subject: template_with_sender_subject"})
+        template_no_sender = template_with_sender_subject.copy({'email_from': '', 'subject':''})
+        template_no_subject = template_no_sender.copy({'subject':''})
+        form = Form(self.env['mail.compose.message'])
+
+        for composition_mode in ['comment','mass_mail']:
+            with self.subTest(msg='subtest per composition_mode', composition_mode=composition_mode):
```

Comment:
as msg will be the same for each run, it does not add much value :) purpose is to give values that distinguish the test (as those are outputed in the logs), so would say only composition_mode should be used as subTest parameters

Reply:
thx for the tip, appreciate it